add handling of mangle_properties regex option in uglifyjs

jrhite opened the pull request:
The uglifyjs tool now allows for specifying regexes when mangling properties. This PR adds wrapper functionality for the feature.

milkisevil commented:
Hey @jrhite, is there any improvement with the technique used in this PR when compared to #352?

jrhite commented:
@milkisevil  @reppners

Sorry, my bad here. I implemented this without checking if there was already a similar PR. They're basically the same. Had I seen the original PR I would have just commented rather than code up a new PR.

I do have a slight preference to my PR for 2 reasons though:

* I changed the `mangleProperties` option to work the same way that the `mangle` option already works. Specifically, as with `mangle` it can either be true/false or an object (which means true) with more detailed options. I like that both `mangle` and `mangleProperties` work in the same way.
* having both `mangleProperties` and `mangleRegex` doesn't make a lot of sense, because `mangleRegex` will just be ignored if `mangleProperties` is false, which might be confusing and produce counter-intuitive behavior.

@XhmikosR @vladikoff, any way we can get some eyes on either this PR or #352 ?

Would be great to get either this PR or that one merged as the regex functionality is pretty handy.

reppners commented:
> I do have a slight preference to my PR for 2 reasons though

I think those are valid reasons. The only thing that was a concern to me was to simply pass through the options to Uglify2 so the API of the grunt task and the underlying Uglify2 lib are the same. I don't know how important that is to the maintainers and users of grunt-uglify. I'm fine with both solutions  :)

davnicwil commented:
+1 for merging this feature when possible (I need it!) and +1 for this approach of nesting mangleRegex inside mangleProperties as an option, makes more sense than having them as two separate options.

XhmikosR commented:
@vladikoff: does this look good to you?

jrhite commented:
Hi @XhmikosR @vladikoff 

Is grunt-contrib-uglify dead or dying? Just curious as to the future of this grunt plugin.

Thanks!

vladikoff commented:
lgtm! @jrhite are you interesting in helping maintain this plugin?

jrhite commented:
@vladikoff thanks for the merge! I am happy to help when and where I can. Not sure about the time commitment but will do what I can!

jrhite commented:
@vladikoff @XhmikosR  thanks guys, happy to see this merged.

I see npm is still on version 0.11.0. What's the process to get the official npm version bumped up?

Thanks!
